neuro_extras/main.py

```diff
@@ -471,19 +435,100 @@ async def _nonstorage_cp(
         ]
     else:
         raise ValueError("Unknown cloud provider")
-    click.echo(f"Running {command} {' '.join(args)}")
+    click.echo(f"Running '{command} {' '.join(args)}'")
     subprocess = await asyncio.create_subprocess_exec(command, *args)
     returncode = await subprocess.wait()
+    if UrlType.get_type(source_url) == UrlType.LOCAL and remove_source:
+        _remove_local(source_url.path)
     if returncode != 0:
         raise click.ClickException("Cloud copy failed")
-    elif UrlType.get_type(source_url) == UrlType.LOCAL:
-        source_path = Path(source_url.path)
-        if remove_source:
-            if source_path.is_dir():
-                dir_util.remove_tree(str(source_path))
-            else:
-                if source_path.exists():
-                    source_path.unlink()
+
+
+async def _extract(src_url: URL, dst_url: URL, rm_src: bool) -> None:
```

Comment:
Sorry while rebasing I slightly refactored this method to make it more readable:
1. rename long names to symmetric `src`/`dst`
2. work with extensions as with a single string not an array

Reply:
Hmm, the logic has changed: now archive `a.b.c.tar.gz` will break the logic.... Reverting it back

Reply:
reverted all these chanes. So the only change after rebase is [this](https://github.com/neuro-inc/neuro-extras/pull/118#discussion_r511858508)